src/domain/search/filters/filter/DateFilter.js

```diff
@@ -48,26 +82,78 @@ class UntranslatedDateFilter extends React.Component {
     onChange(newDate);
   };
 
+  isValidDate = (date, format = 'L') => {
+    return moment(date, format, true).isValid();
+  }
+
+  handleInputChange = (event) => {
+    const value = event.target.value;
+
+    this.setState({ inputValue: value });
+
+    if (value && this.isValidDate(value)) {
+      this.props.onChange(moment(value, 'L', true));
+    }
+  }
+
+  handleInputBlur = () => {
+    this.setState({ inputTouched: true });
+  }
+
   render() {
     const {
       locale,
       label,
       date,
+      name,
+      t,
     } = this.props;
-    const { isOpen } = this.state;
+    const { isOpen, inputValue, inputTouched } = this.state;
+    const inputError = inputTouched && !this.isValidDate(inputValue);
+    const errorMessageId = `${name}-error`;
 
     return (
       <div className="app-DateFilter">
-        <ControlLabel>{label}</ControlLabel>
-        <FormGroup onClick={() => this.setState({ isOpen: !isOpen })}>
+        <FormGroup controlId={name}>
+          <ControlLabel>{label}</ControlLabel>
+          {/* Apply error here so that it doesn't end up under the overlay */}
+          {inputError && (
+            <span
+              className="app-DateFilter__calendar_error"
+              id={errorMessageId}
+              role="alert"
+            >
+                {t('DatePickerControl.form.error.feedback')}
+            </span>
+          )}
           <InputGroup>
-            <InputGroup.Addon className="app-DateFilter__title">
-              <img alt="" className="app-DateFilter__icon" src={iconCalendar} />
-              <span>{moment(date).format('L')}</span>
-            </InputGroup.Addon>
-            <InputGroup.Addon className="app-DateFilter__triangle">
-              <Glyphicon glyph="triangle-bottom" />
-            </InputGroup.Addon>
+            <FormControl
+              aria-describedby={errorMessageId}
+              className="app-DateFilter__input"
+              onBlur={this.handleInputBlur}
+              onChange={this.handleInputChange}
+              onClick={() => this.setState({ isOpen: !isOpen })}
+              onKeyDown={this.handleInputKeydown}
+              type="text"
+              value={inputValue}
+            />
+            <InputGroup.Button>
+              {/* We are setting tab-index as -1 in order to skip */}
+              {/* the button controlling the date picker dropdown. */}
+              {/* Because it's not keyboard accessible, it's better */}
+              {/* to help users avoid it */}
+              <Button
+                aria-hidden="true"
+                className="app-DatePickerControl__button"
+                onClick={this.handleDateButtonClick}
```

Comment:
This method doesn't exist.

Reply:
Added the missing method and a test case. Kind of feel bad for letting that one through :D